dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,519 @@
+
+        # generate verilator simulation cpp code components
+        inputs, outputs = self.generate_cpp_inputs_outputs(tasklet)
+        vector_init = self.generate_cpp_vector_init(tasklet)
+        num_elements = self.generate_cpp_num_elements()
+        internal_state_str, internal_state_var = self.generate_cpp_internal_state(tasklet)
+
+        # add header code do stream
+        sdfg.append_global_code(cpp_code=RTLCodeGen.header_template().format(name=unique_name))
+
+        # add main cpp code to stream
+        callsite_stream.write(contents=RTLCodeGen.main_template().format(name=unique_name,
+                                                                         inputs=inputs,
+                                                                         outputs=outputs,
+                                                                         num_elements=num_elements,
+                                                                         vector_init=vector_init,
+                                                                         internal_state_str=internal_state_str,
+                                                                         internal_state_var=internal_state_var),
+                              sdfg=sdfg,
+                              state_id=state_id,
+                              node_id=node)
+
+    @staticmethod
+    def header_template():
+        """
+            Cpp Header Template
+        """
+        return """\
+// generic includes
+#include <iostream>
+
+// verilator includes
+#include <verilated.h>
+
+// include model header, generated from verilating the sv design
+#include "V{name}.h"
+"""
+
+    @staticmethod
+    def main_template():
+        """
+            Cpp Main Verilator Simulator Template
+        """
+        return """\
+std::cout << "SIM START" << std::endl;
+
+vluint64_t main_time = 0;
+
+// instantiate model
+V{name}* model = new V{name};
+
+// apply initial input values
+model->rst_i = 0;  // no reset
+model->clk_i = 0; // neg clock
+model->valid_i = 0; // not valid
+model->ready_i = 0; // not ready 
+model->eval();
+
+// init vector
+{vector_init}
+
+// reset design
+model->rst_i = 1;
+model->clk_i = 1; // rising
+model->eval();
+model->clk_i = 0; // falling
+model->eval();
+model->rst_i = 0;
+model->clk_i = 1; // rising
+model->eval();
+model->clk_i = 0; // falling
+model->eval();
+
+// simulate until in_handshakes = out_handshakes = num_elements
+bool read_input_hs = false, write_output_hs = false;
+int in_ptr = 0, out_ptr = 0;
+{num_elements}
+
+while (out_ptr < num_elements) {{
+
+    // increment time
+    main_time++;
+
+    // check if valid_i and ready_o have been asserted at the rising clock edge -> input read handshake
+    if (model->ready_o == 1 && model->valid_i == 1){{
+        read_input_hs = true;
+    }} 
+    // feed new element
+    if(model->valid_i == 0 && in_ptr < num_elements){{
+        std::cout << "feed new element" << std::endl;
```

Comment:
```suggestion
```

Reply:
If we want to keep something like this, we need to add a verbosity flag. In many cases the user probably doesn't want every read and write to be printed to the terminal. Even then, it should be more descriptive (_what_ was fed)

Reply:
I totally agree with you. I intend to extend the `dace.conf` section of the rtl codegen by this (see project page, ToDo) and put all the debug output/internal state print statements into an if clause. We decided during the meeting that this is not part of deliverable 1 (but should definitely be done very soon afterwards). Since i want to move most of the simulator functionality to a proper class/functions instead of this flat code, it will be part of this refactorization.